lib/Controller/ControllerProvider.php

```diff
@@ -0,0 +1,37 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\Website\Controller;
+
+use InvalidArgumentException;
+use function get_class;
+use function sprintf;
+
+class ControllerProvider
+{
+    /** @var object[] */
+    private $controllers;
+
+    /**
+     * @param object[] $controllers
+     */
+    public function __construct(array $controllers)
+    {
+        foreach ($controllers as $controller) {
+            $this->controllers[get_class($controller)] = $controller;
+        }
+    }
+
+    /**
+     * @return object
```

Comment:
What PHP version do we require for building website? `composer.json` doesn't specify one, this should be a native return type as of 7.2.

Reply:
I would prefer 7.2 too.

Reply:
7.1 currently.